insights/client/__init__.py

```diff
@@ -603,7 +591,8 @@ def post_update():
     # put this first to avoid conflicts with register
     if config['unregister']:
         pconn = client.get_connection()
-        die(rc=0 if pconn.unregister() else 1)
+        pconn.unregister()
+        sys.exit(42)
```

Comment:
Seems like there should be a difference between "exit now with rc=0" and "exit now with rc=1", but I don't see one with what you've pushed so far.